packages/2d/src/components/Polygon.ts

```diff
@@ -0,0 +1,104 @@
+import {Shape, ShapeProps} from './Shape';
+import {initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+export interface PolygonProps extends ShapeProps {
+  /**
+   * {@inheritDoc Polygon.nSides}
+   */
+  nSides?: SignalValue<number>;
+
+  /**
+   * {@inheritDoc Polygon.angle}
+   */
+  angle?: SignalValue<number>;
+}
+
+export class Polygon extends Shape {
+  /**
+   * Sets the number of sides of the polygon. For example, a value of 6
+   * creates a hexagon.
+   *
+   * @remarks
+   * Note that the polygon is inscribed in a circle defined by the height
+   * and width. If height and width are unequal, the polygon is inscribed
+   * in the resulting ellipse.
+   *
+   * Since the polygon is inscribed in the circle, the actual displayed
+   * height and width may differ somewhat from the bounding rectangle. This
+   * will be particularly noticable if the number of sides is low, e.g. for a
+   * triangle.
+   *
+   *
+   * @example
+   * ```tsx
+   * <Polygon
+   *   x={0}
+   *   y={0}
+   *   width={320}
+   *   height={320}
+   *   nSides={6}
+   *   stroke={'#fff'}
+   *   lineWidth={8}
+   *   fill={'lightseagreen'}
+   * />
+   * ```
+   */
+  @initial(6)
+  @signal()
+  public declare readonly nSides: SimpleSignal<number, this>;
+
+  /**
+   * Changes the starting angle (in degrees) of the polygon.
+   *
+   * @remarks
+   * Use this to change the apparent rotation of the polygon.
+   *
+   * @example
+   * ```tsx
+   * <Polygon
+   *   x={0}
+   *   y={0}
+   *   width={320}
+   *   height={320}
+   *   nSides={6}
+   *   angle={30}
+   *   stroke={'#fff'}
+   *   lineWidth={8}
+   *   fill={'lightseagreen'}
+   * />
+   * ```
+   */
+  @initial(0)
+  @signal()
+  public declare readonly angle: SimpleSignal<number, this>;
+
+  public constructor(props: PolygonProps) {
+    super(props);
+  }
+
+  protected override getPath(): Path2D {
+    const path = new Path2D();
+    const angle = (this.angle() / 360) * 2 * Math.PI;
+    const width = this.width() / 2;
+    const height = this.height() / 2;
+    const nSides = this.nSides();
+    let x0 = 0;
+    let y0 = 0;
+
+    for (let i = 0; i <= nSides; i += 1) {
+      const theta = (i * 2 * Math.PI) / nSides - angle;
+      const x = width * Math.sin(theta);
+      const y = -height * Math.cos(theta);
+      if (i == 0) {
+        path.moveTo(x, y);
+        x0 = x;
+        y0 = y;
+      } else {
+        path.lineTo(x, y);
+      }
+    }
+    path.lineTo(x0, y0);
```

Comment:
There's a potential issue with this approach of "closing" the polygon by manually drawing the final line segment back to the start location: the `lineJoin` property won't get applied to the last and first point of the polygon which–depending on the angle–will produce an ugly gap or overlap.

For what it's worth, this is also a problem with the `Line` component.

I think it might be better to override the `drawShape` method instead of `getPath`. This would allow us to call `context.closePath()` instead of manually drawing the final line segment. This would ensure that the first and last line segments would get joined properly.

Reply:
<img width="343" alt="CleanShot 2023-03-03 at 10 42 40@2x" src="https://user-images.githubusercontent.com/5139098/222686908-cb145327-4031-47a3-9007-e36d2884707f.png">
Here's the issue I'm talking about

Reply:
Sorry, I was overcomplicating the proposed solution a little bit. All that's required is to add a `path.closePath()` call inside the `getPath` method. This also allows us to simplify the method itself a little bit because we no longer have to draw the final line segment ourselves:
```diff
protected override getPath(): Path2D {
    const path = new Path2D();
    const angle = (this.angle() / 360) * 2 * Math.PI;
    const width = this.width() / 2;
    const height = this.height() / 2;
    const nSides = this.nSides();
-   let x0 = 0;
-   let y0 = 0;

-    for (let i = 0; i <= nSides; i += 1) {
+   for (let i = 0; i <= nSides - 1; i += 1) {
      const theta = (i * 2 * Math.PI) / nSides - angle;
      const x = width * Math.sin(theta);
      const y = -height * Math.cos(theta);
      if (i == 0) {
        path.moveTo(x, y);
-       x0 = x;
-       y0 = y;
      } else {
        path.lineTo(x, y);
      }
    }
+   path.closePath();
    return path;
  }
```

<img width="337" alt="CleanShot 2023-03-03 at 10 49 06@2x" src="https://user-images.githubusercontent.com/5139098/222688340-f8e8cca5-1b23-4a5f-97db-c8156f9ad3ab.png">

Reply:
This is great, thanks.  I observed that gap, but didn't know what caused it or how to fix.  Let me investigate.

Reply:
This works great, thank you!